SimDataFormats/CaloHit/src/PCaloHit.cc

```diff
@@ -12,6 +12,12 @@ PCaloHit::PCaloHit(unsigned int id, float eEM, float eHad, float t, int i, uint1
   myEMFraction = (myEnergy <= 0. ? 1. : eEM / myEnergy);
 }
 
+PCaloHit::PCaloHit(unsigned int id, float eEM, float eHad, float t, int i, int iFine, uint16_t d)
+    : myTime(t), myItra(i), myFineItra(iFine), detId(id), myDepth(d) {
+  myEnergy = eEM + eHad;
+  myEMFraction = (myEnergy <= 0. ? 1. : eEM / myEnergy);
```

Comment:
I would avoid type conversion:
  .....  <= 0.f ? 1.f : .......

Reply:
I can change this. This piece of code is a copy-paste from similar methods defined above, so I'll change it there too.

Reply:
OK